controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1527,10 +1524,59 @@ bool master_thread::handle_intel_slave_join(Socket *sd, ieee1905_1::CmduMessageR
     return true;
 }
 
-bool master_thread::handle_non_intel_slave_join(Socket *sd,
-                                                std::shared_ptr<ieee1905_1::tlvWscM1> tlvwscM1,
-                                                std::string bridge_mac, std::string radio_mac,
-                                                ieee1905_1::CmduMessageTx &cmdu_tx)
+/**
+ * @brief parse the radio basic capabilities TLV and store the operating class
+ * in the database as supported channels.
+ * 
+ * @param radio_mac radio mac address
+ * @param radio_caps radio basic capabilities TLV received from the remote agent
+ * @return true on success
+ * @return false on failure
+ */
+bool master_thread::autoconfig_wsc_parse_radio_caps(
+    std::string radio_mac, std::shared_ptr<wfa_map::tlvApRadioBasicCapabilities> radio_caps)
+{
+    // read all operating class list
+    auto operating_classes_list_length = radio_caps->operating_classes_info_list_length();
+    if (operating_classes_list_length > beerocks::message::SUPPORTED_CHANNELS_LENGTH) {
+        LOG(WARNING) << "operating class info list larger then maximum supported channels";
+        operating_classes_list_length = beerocks::message::SUPPORTED_CHANNELS_LENGTH;
+    }
+    for (int oc_idx = 0; oc_idx < operating_classes_list_length; oc_idx++) {
+        std::stringstream ss;
+        auto operating_class_tuple = radio_caps->operating_classes_info_list(oc_idx);
```

Comment:
I'm not sure I understood your proposal. In any case, I guess this should have been a (nitpick) - you're welcome to change this on a followup PR.

Reply:
> I'm not sure I understood your proposal. In any case, I guess this should have been a (nitpick) - you're welcome to change this on a followup PR.

Get rid of this tuple -> tlvf should generate code that returns a pointer instead of a tuple, so you don't need all this crazy std::get<1> stuff.

The part about "get the pointer to the list once from the tuple" is something I don't understand either.